### ISSUE TITLE
Question: source maps 

### ISSUE THREAD

**Issue by gedw99** [apexlang/apex-go#11]
Does this allow to look back at the IDL from the runtime ?

So if you compile a idl , when you’re using it at runtime you can see the line of code from back in the IDL

debugging is then easier.

Maybe source providence is a bette name for this :) 

then with a continuous compiler setup you can see the wrong code, edit it, and get a new wasm module load up. Compiler as a service with quasi “edit and continue” debugging .

**Comment by pkedy** [comment]
Yes, each type of AST Node contains a `Location` property that points back to the line/column and optional `Source` property that includes the source of the Apex.

Go: https://github.com/apexlang/apex-go/blob/main/ast/nodes.go#L30
JS: https://github.com/apexlang/apex-js/blob/main/src/ast/nodes.ts#L31
(I'd have to look but this may not be available in the Wasm module since it returns the model, not the AST.)

This is used in the [VSCode extension](https://github.com/apexlang/vscode-extension) to highlight issues/errors in the Apex file.

> then with a continuous compiler setup you can see the wrong code, edit it, and get a new wasm module load up. Compiler as a service with quasi “edit and continue” debugging .

By "continuous compiler setup" I assume you mean that whenever you update the Apex, that the code generates and if possible, the project is compiled. Right now, the VSCode extension at least runs `apex watch` in the background and automatically regenerates the code.

In a sense, you can treat Apex as code and completely ignore the generated files. Apex _could_ also be configured to run a compiler after codegen. However, I'm wondering if the continuous compilation function should be handled externally, by the IDE or custom setup. If you have thoughts on how this would work, please share them 😄
